frontend/packages/console-shared/src/components/dashboard/status-card/AlertItem.tsx

```diff
@@ -49,20 +55,57 @@ export const StatusItem: React.FC<StatusItemProps> = ({ Icon, timestamp, message
   );
 };
 
-const AlertItem: React.FC<AlertItemProps> = ({ alert, hideLink }) => {
+const AlertItem: React.FC<AlertItemProps> = ({ alert }) => {
   const actionsExtensions = useExtensions<AlertAction>(isAlertAction);
   const action = getAlertActions(actionsExtensions).get(alert.rule.name);
   const { t } = useTranslation();
+  const [resolvedProperty, setResolvedProperty] = React.useState<
+    Omit<AlertAction['properties'], 'alert'> & {
+      actionProperty: string | (() => any);
+    }
```

Comment:
This would be greatly simplified if you were to use `history.push`. You could remove the state and the effect and simply resolve the coderef and execute it `onClick`

Furthermore, why aren't these extensions already resolved using `useResolvedExtensions`?

Reply:
`useResolvedExtensions` resolves all the extensions but we really don't need to resolve all the coderef. Just the one's for which the alert's are active. Hence I am doing it manually.

Reply:
What about doing something like this instead by filtering only on the one action you are interested in?

```
const actionsExtensions = useResolvedExtensions(
    React.useCallback(
      (e) =>
        isAlertAction(e) && e.properties.alert === alert.rule.name,
      [alert.rule.name],
    ),
  );
```